components/tidb_query/src/codec/datum.rs

```diff
@@ -268,9 +271,9 @@ impl Datum {
             Datum::I64(i) => Some(i != 0),
             Datum::U64(u) => Some(u != 0),
             Datum::F64(f) => Some(f.round() != 0f64),
-            Datum::Bytes(ref bs) => {
-                Some(!bs.is_empty() && bs.to_int(ctx, FieldTypeTp::LongLong)? != 0)
-            }
+            Datum::Bytes(ref bs) => Some(
+                !bs.is_empty() && <&[u8] as ConvertTo<i64>>::convert(&bs.as_slice(), ctx)? != 0,
```

Comment:
If this is changed, what about these things in `as_mysql_bool`?

Reply:
Actually, this convert call `to_int(ctx, FieldTypeTp::LongLong)`, so their behavior is same.

I change `to_int(ctx, FieldTypeTp::LongLong)` to `convert` is to avoid `FieldTypeTp:LongLong` which is confused:  why we use `FieldTypeTp::LongLong` but not other tp, we can hide this with     using `convert`